### ISSUE TITLE
Proposal - Use new I2C terminology: change master/slave to controller & target

### ISSUE THREAD

**Issue by awgrover** [ftjuh/AccelStepperI2C#2]
NXP has updated the I2C terminology to "controller" and "target". Can this project migrate to the new terminology?

I'd suggest an update to the README.md first, possibly with footnote. I note that there are some other uses of "master" in the README.md. At the same time, change the release branch to "main".

As a second phase, change the code, including the interface. There are at least a few methods that would need to change name: mark them as deprecated, substitute new ones, update examples/documentation, and plane to retire.

I can submit a pull-request(s).

For the new terminology, please see
https://www.nxp.com/docs/en/user-guide/UM10204.pdf
Table 2.

**Comment by ftjuh** [comment]
Thank you for the link, I wasn't aware of that document.

I find it interesting that your request sounds a lot like a command. Do you care to explain where that comes from?

Also, it seems there are currently [>500 of other projects](https://github.com/search?q=i2c+slave&type=Repositories) at github using the master and/or slave terminology. Why did you pick this one for your request?

**Comment by awgrover** [comment]
I was surprised at the official terminology change, having checked on this only a few months ago (though I could have missed it).  And, I haven't looked at the state of SPI in a while.

I have changed the issue title. Does any of the body of the issue want changing to not sound like a command?

I submitted the issue to this project since I will be using it, and I read the documentation, examples, and examined some of the code. Your project was not singled out for any reason besides the fact that I will be using it. No doubt the other projects should update their terminology, and it is likely I will submit an issue to them as it comes up.

I again extend the offer to do the simple terminology change in the documentation (README), and comments (i.e. no code change). And, as a separate PR, to submit a proposed PR with any code naming changes.

**Comment by ftjuh** [comment]
No, no problem, maybe just an ICC issue. I'm not a native speaker so your request might have sounded a lot harsher in my ears than it was meant. It seems the terminology thing is a heated issue and I was just genuinely interested in your motives.

Concerning your offer I'll be happy for a PR, but the project is in a special phase right now, so please read on before doing anything:

Since the most recent release, I evolved the project into a generic framework for I2C targets. AccelStepperI2C and ServoI2C now are mere modules that can optionally be included into the framework, two new modules, ESP32sensorI2C and PinI2C, have been added, others can easily be added with templates. This project's develop branch has a preview. A target compiled with AccelStepperI2C and ServoI2C modules will be functionally equivalent to this project, from the controller's side nothing will change.

As "AcellStepperI2C" now is too narrow a name for the generic framework, this means a new name ("I2Cwrapper"), which, as a consequence, means a new repository, and a new Arduino library. I'm in the very process of releasing it, in fact I'll try to upload it later, even if some finishing touches might be missing and atm it uses the old terminology.

That also means I don't plan to put any more work into this project, as I2Cwrapper can do anything AccelStepperI2C alone could do, and a lot more on top.

So in my view, it would be best if you reviewed the (completely revised) readme and documentation of the new project, and this project is allowed to retire silently and unchanged.

**Comment by ftjuh** [comment]
I just pushed [the new I2Cwrapper project](https://github.com/ftjuh/I2Cwrapper). Thank you again for your offer, but I think I already did all of the necessary changes while preparing the new project for its initial release.

I would be glad, though, if you, instead of doing the changes yourself, were willing to give it an overall critical review.

Please note that it's probably best to remove the old library folder, so that the Arduino environment will pick up the correct library versions. Also, the new library is not submitted to the Arduino library manager yet, so it will have to be installed manually for the moment.
